type.go

```diff
@@ -640,6 +730,48 @@ func (t floatType) AssignValue(dst reflect.Value, src Value) error {
 	switch dst.Kind() {
 	case reflect.Float32, reflect.Float64:
 		dst.SetFloat(float64(v))
+	case reflect.Pointer:
+		underlying := reflect.TypeOf(dst).Elem().Kind()
+		switch underlying {
+		case reflect.Float32:
+			dst.Set(reflect.ValueOf(&v))
+		case reflect.Float64:
+			val := float64(v)
+			dst.Set(reflect.ValueOf(&val))
```

Comment:
Same as for the int cases, I would like to ask if we could unify the code paths for value and pointer assignments so we don't have to repeat the rules.

Reply:
refactored the code.